behave/runner.py

```diff
@@ -918,10 +918,21 @@ def load_hooks(self, filename=None):
     def load_step_definitions(self, extra_step_paths=None):
         if extra_step_paths is None:
             extra_step_paths = []
+
         # -- Allow steps to import other stuff from the steps dir
         # NOTE: Default matcher can be overridden in "environment.py" hook.
+        # Collect directories containing step files
         steps_dir = os.path.join(self.base_dir, self.config.steps_dir)
-        step_paths = [steps_dir] + list(extra_step_paths)
+        step_paths = set()  # Use a set to avoid duplicate paths
```

Comment:
I'm not sure if that couldn't lead to unpleasant side-effects. Could you add a test that proves that two subdirectories having the same name but on different directory levels don't cancel out each other?

Reply:
I added a test to verify that they don't cancel each other out, and the test passed without issues.